src/nvim/api/vim.c

```diff
@@ -2996,3 +2996,37 @@ void nvim_set_decoration_provider(Integer ns_id, DictionaryOf(LuaRef) opts,
 error:
   decor_provider_clear(p);
 }
+
+/// @param opts Optional parameters; only valid key is the `docs` boolean.
+///
+/// @returns |api-metadata| map (Dictionary) with an optional additional `docs`
+///          key containing function documentation as a map (Dictionary) of the
+///          form { "nvim_win_set_option": {...}, "nvim_win_get_option": {...}, ... }
+Dictionary nvim_get_api(Dictionary opts, Error *err)
+  FUNC_API_SINCE(7) FUNC_API_FAST
+{
+  if (opts.size > 1) {
+    api_set_error(err, kErrorTypeValidation, "too many options passed");
+    return (Dictionary)ARRAY_DICT_INIT;
```

Comment:
This is just so I can assume further down that there's only one key, and so don't have to iterate over the dictionary but can just get `opts.items[0]`; I could do a `for` loop though like in `nvim_get_context` if that's preferred?

Reply:
You could just replace `if (opts.size > 0) {` with the for loop header (one line) and then remove this check tho. it is also more consistent with other single-opt functions and easier to extend later.

Reply:
Done.